pkg/nvcdi/driver-nvml.go

```diff
@@ -99,21 +99,34 @@ func (l *nvcdilib) NewDriverLibraryDiscoverer(version string) (discover.Discover
 
 	var discoverers []discover.Discover
 
-	driverDotSoSymlinksDiscoverer := discover.WithDriverDotSoSymlinks(
-		libraries,
-		version,
-		l.nvidiaCDIHookPath,
-	)
-	discoverers = append(discoverers, driverDotSoSymlinksDiscoverer)
+	if l.HookIsSupported(HookCreateSymlinks) {
```

Comment:
For this hook specifically (and the update-ldcache hook too), there are many places in the code where we are injecting this hook. As such this is not going to be sufficient to prevent their injection. This was why I suggested that handling these hooks is out of scope for this PR.